tests/waku_filter_v2/test_waku_client.nim

```diff
@@ -9,720 +9,941 @@ import
-    ) {.async, gcsafe, closure.} =
-      pushHandlerFuture.complete((pubsubTopic, message))
-
-    pubsubTopic = DefaultPubsubTopic
-    contentTopic = DefaultContentTopic
-    contentTopicSeq = @[DefaultContentTopic]
-    serverSwitch = newStandardSwitch()
-    clientSwitch = newStandardSwitch()
-    wakuFilter = await newTestWakuFilter(serverSwitch)
-    wakuFilterClient = await newTestWakuFilterClient(clientSwitch)
-    wakuFilterClient.registerPushHandler(messagePushHandler)
-    
-    await allFutures(serverSwitch.start(), clientSwitch.start())
-    serverRemotePeerInfo = serverSwitch.peerInfo.toRemotePeerInfo()
-    clientPeerId = clientSwitch.peerInfo.toRemotePeerInfo().peerId
-  
-  asyncTeardown:
-    await allFutures(wakuFilter.stop(), wakuFilterClient.stop(), serverSwitch.stop(), clientSwitch.stop())
-
-  suite "Subscriber Ping":
-    asyncTest "Active Subscription Identification":
-      # Given
-      let 
-        subscribeResponse = await wakuFilterClient.subscribe(
+  suite "MessagePushHandler - Void":
+    var serverSwitch {.threadvar.}: Switch
+    var clientSwitch {.threadvar.}: Switch
+    var wakuFilter {.threadvar.}: WakuFilter
+    var wakuFilterClient {.threadvar.}: WakuFilterClient
+    var serverRemotePeerInfo {.threadvar.}: RemotePeerInfo
+    var pubsubTopic {.threadvar.}: PubsubTopic
+    var contentTopic {.threadvar.}: ContentTopic
+    var contentTopicSeq {.threadvar.}: seq[ContentTopic]
+    var clientPeerId {.threadvar.}: PeerId
+    var messagePushHandler {.threadvar.}: FilterPushHandler
+    var pushHandlerFuture {.threadvar.}: Future[(string, WakuMessage)]
+
+    asyncSetup:
+      pushHandlerFuture = newPushHandlerFuture()
+      messagePushHandler = proc(
+        pubsubTopic: PubsubTopic, message: WakuMessage
+      ): Future[void] {.async, closure, gcsafe.} =
+        pushHandlerFuture.complete((pubsubTopic, message))
+
+      pubsubTopic = DefaultPubsubTopic
+      contentTopic = DefaultContentTopic
+      contentTopicSeq = @[DefaultContentTopic]
+      serverSwitch = newStandardSwitch()
+      clientSwitch = newStandardSwitch()
+      wakuFilter = await newTestWakuFilter(serverSwitch)
+      wakuFilterClient = await newTestWakuFilterClient(clientSwitch)
+
+      await allFutures(serverSwitch.start(), clientSwitch.start())
+      wakuFilterClient.registerPushHandler(messagePushHandler)
+      serverRemotePeerInfo = serverSwitch.peerInfo.toRemotePeerInfo()
+      clientPeerId = clientSwitch.peerInfo.toRemotePeerInfo().peerId
+      
+    asyncTeardown:
+      await allFutures(wakuFilter.stop(), wakuFilterClient.stop(), serverSwitch.stop(), clientSwitch.stop())
+
+    suite "Subscriber Ping":
+      asyncTest "Active Subscription Identification":
+        # Given
+        let 
+          subscribeResponse = await wakuFilterClient.subscribe(
+            serverRemotePeerInfo, pubsubTopic, contentTopicSeq
+          )
+        require:
+          subscribeResponse.isOk()
```

Comment:
For that kind of validation, I think it is better to do the next so that we have a better detail in case of failure:
`assert subscribeResponse.isOk(), $subscribeResponse.error`

That applies elsewhere a `Result` is being checked.

Reply:
Agree, must've slipped by when changing. 
I'm actually moving away from require, as if a require doesn't work, it makes the whole test suite fail, so I'd rather use checks or asserts.